pkg/auth/token_verifier.go

```diff
@@ -158,3 +158,18 @@ type openIDMetadata struct {
 	SubjectTypes  []string `json:"subject_types_supported"`
 	SigningAlgs   []string `json:"id_token_signing_alg_values_supported"`
 }
+
+func ValidateEventAuthHeader(ctx context.Context, request *http.Request) error {
+	features := feature.FromContext(ctx)
+	if features.IsOIDCAuthentication() {
```

Comment:
IMO we should keep this function independent from the feature config, and should the caller decide if they want it or not.

Reply:
Ah good to know - that's my bad for suggesting this change then!

Reply:
No worries. This is just my opinion. If there are valid arguments for the other way, we should discuss it.
I thought only to keep it independent, so the caller has the option to decide. E.g. the `EnsureOIDCServiceAccountExistsForResource()` is also only called, when the feature is enabled (checked by the caller / reconciler)